### PR TITLE
Add royalties to collection mint

### LINKED ISSUE #8130: add royalty to collection creation
@vikiival  added the ability to set royalty on collection (kusama / polkadot assethub)
this royalty will be the default one for all the NFTs in the collection

https://github.com/kodadot/stick/pull/148

need to use this new capability in FE

1. add the fields to the collection mint form
2. `set_attribute` to put the royalty on chain

### PR DESCRIPTION
**Pull request by brano-hozza**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #8130 
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16DkEx1FzrKqTLPqTJNsSosKJ2UGbUVoLEhiGFFV8cYFpfj1)

#### Community participation

- [x] [Are you at KodaDot Ecosystem Telegram?](https://t.me/kodadot_eco)

## Screenshot 📸

- [x] My fix has changed collection mint form on UI; a screenshot is best to understand changes for others.
image

### CODE REVIEW

**Review by roiLeo — COMMENTED** [review]
Hey, small stuff.

It's missing on Create Nft to display default collection royalty when has been set on collection.

File: composables/transaction/mintCollection/transactionMintCollectionStatemine.ts [hunk 1]
```diff
@@ -56,8 +59,28 @@ export async function execMintCollectionStatemine({
...
+    )
+    const setRoyaltyRecipient = api.tx.nfts.setAttribute(
+      nextId,
+      null,
+      'CollectionOwner',
+      'recipient',
+      royalty.address,
+    )
+    royaltyArgs.push(setRoyaltyAmount, setRoyaltyRecipient)
+  }
```
[the remark this answers is not in the document]
Comment (brano-hozza) [reply composables/transaction/mintCollection/transactionMintCollectionStatemine.ts:79]: I did this after chatting with @vikiival so I am not sure

File: components/create/CreateCollection.vue [hunk 2]
```diff
@@ -105,6 +105,14 @@
         </div>
       </NeoField>
 
+      <!-- royalty -->
+      <NeoField v-if="!isRmrk">
```
[the remark this answers is not in the document]
Comment (brano-hozza) [reply components/create/CreateCollection.vue:109]: It's the same as in `CreateNFT` component, should it be different?

File: composables/transaction/mintCollection/transactionMintCollectionStatemine.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (vikiival) [reply composables/transaction/mintCollection/transactionMintCollectionStatemine.ts:79]: There is no such thing as `set_collection_attribute`


Screenshot 2024-01-23 at 19 22 51

File: components/create/CreateCollection.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (vikiival) [reply components/create/CreateCollection.vue:109]: RoiLeo is right, collection royalty only works on assethub

**Comment by vikiival** [comment]
> It's missing on Create Nft to display default collection royalty when has been set on collection.

How would you do it? 

Because I would hide set royalty on create NFT in asset hub, 
just to make info text

File: composables/transaction/mintCollection/transactionMintCollectionStatemine.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (roiLeo) [reply composables/transaction/mintCollection/transactionMintCollectionStatemine.ts:79]: nvm I was looking at https://github.com/paritytech/polkadot-sdk/blob/a817d310bb5c88267e57eb41fea3f9f56124f8e0/substrate/frame/nfts/src/impl_nonfungibles.rs#L296

**Comment by brano-hozza** [comment]
So which one?

**Comment by vikiival** [comment]
I think roiLeo has good solution for royalties handling.
I have opened issue #9120 because the scope of #8130 was to add this input to collection.

So if you want @brano-hozza you can do follow-up issue.

File: composables/transaction/mintCollection/transactionMintCollectionStatemine.ts [hunk 3]
```diff
@@ -56,8 +59,28 @@ export async function execMintCollectionStatemine({
...
+
   const arg = [
     [
+      ...royaltyArgs,
```
Comment (vikiival) [inline composables/transaction/mintCollection/transactionMintCollectionStatemine.ts:83]: This has to be the last arg :) (or minimally afer create Collection)

**Review by vikiival — CHANGES_REQUESTED** [review]
~ One thingie

**Review by vikiival — APPROVED** [review]

**Review by roiLeo — APPROVED** [review]

**Review by preschian — APPROVED** [review]
lgtm (not tested)

**Comment by yangwao** [comment]
Thanks!
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 6.7 USD/DOT ~ 7.463 $DOT
🧗 [16DkEx1FzrKqTLPqTJNsSosKJ2UGbUVoLEhiGFFV8cYFpfj1](https://polkadot.subscan.io/account/16DkEx1FzrKqTLPqTJNsSosKJ2UGbUVoLEhiGFFV8cYFpfj1)
🔗 [0x1080b1af0be627cbf45bdbe0209108ddade731c37fc6d9184cd447810cff4240](https://polkadot.subscan.io/extrinsic/0x1080b1af0be627cbf45bdbe0209108ddade731c37fc6d9184cd447810cff4240)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**